### ISSUE TITLE
com.sun.jdi.ObjectReference.setValue implementation is not spec complaint

### ISSUE THREAD

**Issue by alexmenkov** [eclipse-jdt/eclipse.jdt.debug#183]
Specification of the JDI method ObjectReference.setValue has changed in JDK 20 to require the field be non-final:
https://bugs.openjdk.org/browse/JDK-8280798 - the fix
https://bugs.openjdk.org/browse/JDK-8298799 - release note
Eclipse JDI implementation allows modification of final instance fields, so it becomes not compliant with the updated spec

**Comment by iloveeclipse** [comment]
@SarikaSinha : I don't remember what our conclusion was on this (not sure we had one): are we going to completely remove the support or just keep it "as is" - by default, since f91b69638f6c81157d6929e65fa9477a09f0b083 we show a warning that this is not a good idea to change final field values?

**Comment by iloveeclipse** [comment]
@alexmenkov : I assume JVM will refuse to perform this operation after https://bugs.openjdk.org/browse/JDK-8280798 ?

if so, we might conditionally disable this "feature" in Eclipse (independently on the preference value) if we see debuggee running Java 20+?

@trancexpress : FYI.

**Comment by SarikaSinha** [comment]
We wanted to give the message on edit rather than save.
So now we need to se how can we disable edit/save for final values for Projects with compliance level >= 20.

**Comment by alexmenkov** [comment]
> I assume JVM will refuse to perform this operation after https://bugs.openjdk.org/browse/JDK-8280798 ?
@iloveeclipse No. There is no changes in JVM, just JDI spec change. So this "feature" will work in Eclipse debugger with jdk20+ (if you decide to keep it)
